### PR TITLE
[system-probe] Setup tracer instance for tests

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Sets up the tracer instance for tests in a uniform way, registering a client for connection state.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/tracer_usm_linux_test.go [hunk 1]
```diff
@@ -243,15 +229,11 @@ func testHTTPSLibrary(t *testing.T, fetchCmd []string, prefechLibs []string) {
 	cmd := append(fetchCmd, targetURL)
 	requestCmd := exec.Command(cmd[0], cmd[1:]...)
 	var out []byte
```
Comment (guyarb) [inline pkg/network/tracer/tracer_usm_linux_test.go:231]: nit: can be removed

**Review by guyarb — APPROVED** [review]
LGTM, left one small suggestion

**Review by akarpz — APPROVED** [review]

File: pkg/network/tracer/tracer_linux_test.go [hunk 2]
```diff
@@ -483,12 +460,7 @@ func TestConntrackDelays(t *testing.T) {
 func TestTranslationBindingRegression(t *testing.T) {
 	setupDNAT(t)
 
-	tr, err := NewTracer(testConfig())
-	require.NoError(t, err)
-	defer tr.Stop()
-
-	// Warm-up tracer state
```
Comment (mbakht) [inline pkg/network/tracer/tracer_linux_test.go:490]: Is this warm-up no longer needed?

**Review by mbakht — APPROVED** [review]

File: pkg/network/tracer/tracer_linux_test.go (hunk 2 above)
Comment (hmahmood) [reply pkg/network/tracer/tracer_linux_test.go:490]: Its done inside `setupTracer` now.

**Pull request merged**